### PR TITLE
Fix a bug adding unrelated candidates to the redirect_list

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fix unrelated candidates can be added to the internal redirect list. For example
```
- legacymodule:
  collections:
    - ns.coll
```
will end up with `ns.coll.legacymodule` in the redirect list even though the module in library/ is used.

Besides ensuring the resolved name is present, the original issue that line was really meant to solve was that `ansible.builtin` would be in the redirect chain - e.g., instead of `['aws_s3', 'amazon.aws.aws_s3']` it would be `['aws_s3', 'ansible.builtin.aws_s3', 'amazon.aws.aws_s3']` so that module_defaults action groups worked uniformly whether the module as called as `ansible.builtin.module` or `module` (assuming `module` is equivalent). It looks like this also needs to be done for non-group module_defaults with ansible.legacy it seems, since action plugins like `gather_facts` have replaced the default list of modules to all use FQCN. Now there's no way to tell if the FQCN was explicitly requested or if shortname matching should still work. For example, this should still work:
```
- gather_facts:
  module_defaults:
    setup:  # (maybe a bad example, as `setup` isn't present, but `smart`; anything in CONNECTION_FACTS_MODULES would have this problem though)
      ...
```
~But even after #72918 is fixed, the module_defaults entry would need to be `ansible.legacy.setup` for ^ example. Still looking at how to add the legacy/builtin FQCNs back (or drop the redirect list)...~

Going to try to keep these changes concise and deal with `ansible.legacy` separately since it hasn't been part of the redirect list yet and I think this would be good to backport.

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

**Comment by s-hertel** [comment]
TL;DR The line being removed in this PR was handling two different scenarios. 1. including the fq name responsible for the redirect in the redirect list and 2. ensuring the resolved fq name is in the redirect list. But it was also included unrelated candidate names.

File: test/integration/targets/collections_internal_redirect/collections/ansible_collections/testns/testcoll/plugins/modules/ping.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python
```
Comment (nitzmahone) [inline test/integration/targets/collections_internal_redirect/collections/ansible_collections/testns/testcoll/plugins/modules/ping.py]: any reason this needs to be a "real" copy of the ping module vs just a dummy module? (see the collections integration tests, there's a ton of dummy modules in there)

File: test/integration/targets/collections_internal_redirect/collections/ansible_collections/ns/coll/plugins/action/test_redirect.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Copyright: (c) 2021, Ansible Project
```
Comment (nitzmahone) [inline test/integration/targets/collections_internal_redirect/collections/ansible_collections/ns/coll/plugins/action/test_redirect.py]: this has a lot of overlap with https://github.com/ansible/ansible/blob/devel/test/integration/targets/collections/collection_root_user/ansible_collections/testns/testcoll/plugins/action/plugin_lookup.py - even if you want to keep the dedicated action, this whole thing probably doesn't need to be a dedicated test target. Just add the custom bits you need under `collection_root_user` (as that's kind of the main one for collections). I actually considered making `plugin_lookup` a supported core action at one point- it might still be worth thinking about...

**Review by nitzmahone — CHANGES_REQUESTED** [review]
Fix itself LGTM.
* needs changelog
* tests should probably be jammed into the existing kitchen-sink `collections` integration test target to avoid duplication.

A lot of these things are already covered in there, so you might want to just expose the redirect list on the result of `plugin_lookup` and assert a few extra things in the existing redirect tests there. It is a big test target, so I understand reticence to find the right place. Let me know if I can help you understand or restructure that to make it easier to find things, as it definitely grew organically over the collections dev period and is a mess to find things.

File: test/integration/targets/collections_internal_redirect/collections/ansible_collections/ns/coll/plugins/action/test_redirect.py (hunk 2 above)
Comment (s-hertel) [reply test/integration/targets/collections_internal_redirect/collections/ansible_collections/ns/coll/plugins/action/test_redirect.py]: Thanks, I hadn't noticed that action plugin! The diff is much smaller now.

**Review by nitzmahone — APPROVED** [review]
Good to merge once CI is green- thanks!

**Comment by s-hertel** [comment]
@nitzmahone Thanks for reviewing
